src/tests/unit/tests/reports/components/new-tab-link-confirmation-dialog.test.tsx

```diff
@@ -0,0 +1,172 @@
+            testSubject2.find(NewTabLink).prop('id'),
+        );
+    });
+
+    it('contains NewTabLink with forwarded props', () => {
+        const props = {
+            href: 'test-href',
+            title: 'test-title',
+        };
+        const child = <div>test-child</div>;
+
+        const testSubject = shallow(
+            <NewTabLinkWithConfirmationDialog {...props}>{child}</NewTabLinkWithConfirmationDialog>,
+        );
+
+        const testSubjectNewTabLink = testSubject.find(NewTabLink);
+
+        expect(testSubjectNewTabLink.prop('href')).toEqual(props.href);
+        expect(testSubjectNewTabLink.prop('title')).toEqual(props.title);
+        expect(testSubjectNewTabLink.contains(child)).toBe(true);
+    });
+
+    describe('onClickHandler', () => {
+        const targetPageLinkMock = Mock.ofType<HTMLElement>();
+
+        type GetElementById = typeof Document.prototype.getElementById;
+
+        let getElementByIdMock: IMock<GetElementById>;
+        let confirmMock: IMock<ConfirmType>;
+
+        let originalGetElementById: GetElementById;
+        let originalConfirm: ConfirmType;
+
+        beforeEach(() => {
+            originalGetElementById = document.getElementById;
+            originalConfirm = window.confirm;
+
+            getElementByIdMock = Mock.ofType<GetElementById>(undefined, MockBehavior.Strict);
+            confirmMock = Mock.ofType<ConfirmType>(undefined, MockBehavior.Strict);
+
+            getElementByIdMock
+                .setup(handler => handler(It.isAnyString()))
+                .returns(() => targetPageLinkMock.object);
+
+            document.getElementById = getElementByIdMock.object;
+            window.confirm = confirmMock.object;
+        });
+
+        afterEach(() => {
+            document.getElementById = originalGetElementById;
+            window.confirm = originalConfirm;
+        });
+
+        it('is added to the link', () => {
+            getElementByIdMock.reset();
+
+            targetPageLinkMock
+                .setup(link => link.addEventListener('click', It.is(isFunction)))
+                .verifiable(Times.once());
+
+            const testSubject = shallow(<NewTabLinkWithConfirmationDialog />);
+
+            const linkId = testSubject.find(NewTabLink).prop('id');
+            getElementByIdMock
+                .setup(handler => handler(linkId))
+                .returns(() => targetPageLinkMock.object)
+                .verifiable(Times.once());
+
+            const generatedScript = testSubject.find('script').render().html();
+
+            // tslint:disable-next-line: no-eval
+            eval(generatedScript);
+
+            targetPageLinkMock.verifyAll();
+            getElementByIdMock.verifyAll();
+        });
+
+        it('calls confirm function with expected text', () => {
+            const eventMock = Mock.ofType<Event>();
+            eventMock.setup(event => event.preventDefault());
+
+            const expectedConfirmationMessage =
```

Comment:
Would it be worth it to try using snapshot for this value? Maybe on the mock setup code on line 105, something like:
```typescript
confirmMock
    .setup(handler => handler(It.isAnyString())
    .callback(message => expect(message).toMatchSnapshot())
    .verifiable(Times.once());
```

Reply:
Updated according to suggestion